### ISSUE TITLE
Scala.js support

### ISSUE THREAD

**Issue by ngbinh** [julien-truffaut/Monocle#230]
There is nothing wrong with https://github.com/japgolly/Monocle but I think it is nicer in the long run to have scala.js support upstream here.

Problems:
- [ ] No official support for scalaz scala.js build yet but @japgolly  https://github.com/japgolly/scalaz should do fine for now.

**Comment by ngbinh** [comment]
looks like @japgolly has already done half of the work here: https://github.com/japgolly/Monocle/commit/0b32fed84a0077e86717d630bc6fbafb2e9445a5

**Comment by julien-truffaut** [comment]
from what I got, it seems that https://github.com/InTheNow/catalysts simplifies the integration with scala.js

**Comment by ngbinh** [comment]
there is also no official scala.js build for `scalaz` yet. Is there a way to have `monocle-core` without `scalaz`?

**Comment by julien-truffaut** [comment]
I think @japgolly released scalaz 7.2 for scala.js

**Comment by japgolly** [comment]
Yeah I did. (Hopefully the last unofficial release - I'll try to get official support in for 7.2.1).

Maybe this is what @ngbinh is getting at too: I'm hesitant to upgrade my projects to Scalaz 7.2 because Monocle is tied to 7.1. I think it might be an idea to make monocle-core dependency-free and move the scalaz stuff into a monocle-scalaz module.

**Comment by ngbinh** [comment]
yes, official support is what I am getting at. And a separate `monocle-core` from `monocle-scalaz` and possibly `monocle-cats` will be awesome!

**Comment by julien-truffaut** [comment]
I don't think we can have a useful `monocle-core` without scalaz or cats. For example, a traversal is defined as:

```scala
def traversal[F[_]: Applicative, S, T, A, B](f: A => F[B])(s: S): F[T]
```

**Comment by xuwei-k** [comment]
scalaz 7.2.1 (first official scalajs support) released :)
- https://github.com/scalaz/scalaz/wiki/7.2.1
- https://github.com/scalaz/scalaz/commit/d646388c573416f837b6afbb50ce079b8696391a
- https://groups.google.com/d/topic/scalaz/NPOdiT9lv-E/discussion

**Comment by ngbinh** [comment]
Awesome! Thanks @xuwei-k

**Comment by julien-truffaut** [comment]
Cool it should simplify official scala.js support for monocle.

**Comment by japgolly** [comment]
Submitted PR #335. Let's see how Travis goes.

Also FYI, I published 1.2.0-2 under my fork to use Scalaz 7.2.1 official.
